src/test/java/org/jfrog/hudson/util/converters/ArtifactoryBuilderConverterTest.java

```diff
@@ -0,0 +1,59 @@
+package org.jfrog.hudson.util.converters;
+
+import hudson.ExtensionList;
+import jenkins.model.Jenkins;
+import org.hamcrest.Matchers;
+import org.jfrog.hudson.ArtifactoryBuilder;
+import org.jfrog.hudson.JFrogPlatformInstance;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+
+public class ArtifactoryBuilderConverterTest {
+
+    private static final String SERVER_ID = "1";
+    private static final String PLATFORM_URL = "https://jfrog.io";
+    private static final String ARTIFACTORY_URL = "https://jfrog.io/artifactory";
+
+    @ClassRule
+    public static JenkinsRule jenkins = new JenkinsRule();
+
+    @Test
+    public void testConvertToArtifactoryServers_pre_3_10_6() throws IOException {
+        ArtifactoryBuilder.DescriptorImpl testBuilder = ExtensionList.lookupSingleton(ArtifactoryBuilder.DescriptorImpl.class);
+        File testFile = new File(getClass().getResource("/converters/config-3.10.6-1.xml").getFile());
```

Comment:
By convention, Jenkins migration test logic like this typically uses the `org.jvnet.hudson.test.recipes.LocalData` annotation, but this technically works.

Reply:
https://github.com/jfrog/jenkins-artifactory-plugin/pull/455#discussion_r642029632